bentoml/cli/bento_service.py

```diff
@@ -477,20 +483,20 @@ def containerize(bento, push, tag, username, password):
 
         _echo(f"Found Bento: {bento_service_bundle_path}")
 
-        name, version = get_name_version_from_tag(bento)
-        # use tag name and version where applicable
-        if tag is not None:
-            name, v = get_name_version_from_tag(tag)
-            if v is not None:
-                version = v
-
-        name, version = make_bento_name_docker_compatible(name, version)
+        bento_service_metadata_pb = load_bento_service_metadata(
+            bento_service_bundle_path
+        )
+        name, version = make_bento_name_docker_compatible(
```

Comment:
This behavior might be confusing to the end-user, I'd prefer to avoid implicitly converting the command argument they provide. For example, I write the following bash script to automate my deployment process:

```bash
saved_path="/Users/chaoyu/bentoml/repository/IrisClassifier/20200701222443_EC5EAB"
image_name="my-image-something-that-is-very-long..........."
bentoml containerize $saved_path -t $image_name
docker run $image_name
``` 
I will get an `Unable to find image` error due to the fact that the image name was modified implicitly.

I'm inclined to go with a simpler approach and only do the conversion if the value is automatically set from the BentoService metadata:
```python
...
bs_metadata = load_bento_service_metadata(bento_service_bundle_path)

_echo(f"Building docker image for BentoService {bs_metadata.name}:{bs_metadata.version}")

if not tag:
   tag = "{to_valid_docker_image_name(bs_metadata.name)}:{to_valid_docker_image_tag(bs_metadata.version)}"

if ":" not in tag:  # user have specified image name but not image tag in -t
   version= to_valid_docker_image_tag(bs_metadata.version)
   tag = f"{tag}:{version}"

_echo("Building docker image with tag: {tag}")

docker_api = docker.APIClient()
...
```